android/app/src/main/java/fi/fmi/mobileweather/LargeForecastWidgetProvider.java

```diff
@@ -65,31 +67,31 @@ protected void setWidgetData(JSONArray announcementsJson, SharedPreferencesHelpe
                     widgetRemoteViews.setTextViewText(R.id.locationRegionTextView, region);
                 }
 
+                RemoteViews timeStep = new RemoteViews(context.getPackageName(), R.layout.forecast_timestep);
 
                 // time at the selected location
                 String localTime = forecast.getString("localtime");
                 String temperature = forecast.getString("temperature");
-                String weatherSymbol = forecast.getString("smartSymbol");
-
-                // get timeTextView0 or timeTextView1 etc. based on i from widgetRemoteViews
-                int timeTextViewId = context.getResources().getIdentifier("timeTextView" + i, "id", context.getPackageName());
-                int temperatureTextViewId = context.getResources().getIdentifier("temperatureTextView" + i, "id", context.getPackageName());
-                int weatherIconImageViewId = context.getResources().getIdentifier("weatherIconImageView" + i, "id", context.getPackageName());
+                int weatherSymbol = forecast.getInt("smartSymbol");
 
                 // ** set the time, temperature and weather icon
 
                 String formattedTime = getFormattedWeatherTime(localTime);
-                widgetRemoteViews.setTextViewText(timeTextViewId, formattedTime);
+                timeStep.setTextViewText(R.id.timeTextView, formattedTime);
 
                 temperature = addPlusIfNeeded(temperature);
-                widgetRemoteViews.setTextViewText(temperatureTextViewId, temperature + "°");
+                timeStep.setTextViewText(R.id.temperatureTextView, temperature + "°");
 
                 int drawableResId = context.getResources().getIdentifier("s_" + weatherSymbol + (theme.equals(LIGHT) ? "_light" : "_dark"), "drawable", context.getPackageName());
-                widgetRemoteViews.setImageViewResource(weatherIconImageViewId, drawableResId);
-            }
+                timeStep.setImageViewResource(R.id.weatherIconImageView, drawableResId);
+                timeStep.setContentDescription(R.id.weatherIconImageView, getSymbolTranslation(weatherSymbol));
 
-            // Update time TODO: should be hidden for release
-            widgetRemoteViews.setTextViewText(R.id.updateTimeTextView, DateFormat.getTimeInstance().format(new Date()));
+                if (i == TIMESTEP_COUNT - 1) {
+                    timeStep.setViewVisibility(R.id.forecastBorder, View.GONE);
+                }
+
+                widgetRemoteViews.addView(R.id.hourForecastRowLayout, timeStep);
+            }
 
```

Comment:
I just tested this in an emulator.
The large widget does not look completely OK now:
![Screenshot 2024-12-12 at 8 12 31](https://github.com/user-attachments/assets/a303dd66-e5a2-4fed-bd83-a228fe3ef744)

- The lower edge of location strings is a bit cut off.
- Too big right margin after time steps row
- Too small margin between location text and time row (same problem also earlier)
- A bit too small margins between time, weather icons and temperature (same problem also earlier)

Reply:
When location strings are shorter that right margin looks OK:
![Screenshot 2024-12-12 at 8 42 11](https://github.com/user-attachments/assets/f1f40446-6ffd-4fce-a5af-3cd0f4bc9f8b)

Reply:
That right margin problem with long location strings seems to be also in the earlier implementation.
If needed it can be fixed later.